refactor: project url uses namespace and name instead of id

vfried opened the pull request:
This addresses #167 

Old URLs that where using the project id will be rerouted to the corresponding namespace/project-name url is possible.

There could be some edge cases that overlap with the new routing, priority will be given to the new schema, i think this edge cases don't exist in reality in renku.

Test Deployment:
https://virginiatest.dev.renku.ch/

Since this traverses all of the projects routing I would appreciate you testing it, if you find some error let me know :)

lorenzo-cavazzi commented on src/project/Project.js:
```diff
@@ -50,11 +50,29 @@ class View extends Component {
+    const potentialProjectId = this.props.projectSlug.split('/')[0];
+    const potentialRoute = this.props.projectSlug.split('/')[1];
+
+    if(!available && !isNaN(potentialProjectId) && routes.indexOf(potentialRoute) > 0){
```
I guess it's `>= 0` instead of `>`

vfried replied:
Actually, there was a check missing, in case the projectSlug is undefined, meaning that is actually a projectId

vfried replied:
But well spotted anyways

lorenzo-cavazzi commented:
I am trying to navigate on all pages.

I get an error when I try to visualize a notebook file while in "lineage view", like in this one
https://virginiatest.dev.renku.ch/projects/lorenzo.cavazzi.tech/zurich-bikes-tutorial/files/lineage/notebooks/zhbikes-notebook.ipynb

lorenzo-cavazzi commented:
Another small issue. When I try to use an old project url, I get an error for the overview page.
E.G. https://virginiatest.dev.renku.ch/projects/95/ (with or without the final slash) doesn't work, but any sub url of that project works fine, like https://virginiatest.dev.renku.ch/projects/95/settings.

vfried commented:
> I am trying to navigate on different pages.
> 
> I get an error when I try to visualize a notebook file while in "lineage view", like in this one
> https://virginiatest.dev.renku.ch/projects/lorenzo.cavazzi.tech/zurich-bikes-tutorial/files/lineage/notebooks/zhbikes-notebook.ipynb

I just tried to navigate to the graph of a file with format .ipynb on de.renku.ch and it also fails. This is not a problem of the file path but i'm not sure where this problem comes from.

vfried commented:
> E.G. https://virginiatest.dev.renku.ch/projects/95/ (with or without the final slash) doesn't work, but any sub url of that project works fine, like https://virginiatest.dev.renku.ch/projects/95/settings.

Fixed

vfried commented:
Yes, sure... wouldn't it be feat(project)?

lorenzo-cavazzi commented:
This has already been approved, but I found a couple of edge cases that may be confusing for the users

* Endless spinning wheel when trying to access a non existing project by id -- or a private project as well (https://virginiatest.dev.renku.ch/projects/9500/)
![Screenshot from 2019-08-27 13-29-48](https://user-images.githubusercontent.com/43481553/63767974-50bf2380-c8cf-11e9-9ac9-3ca963ff83e4.png)

* Empty project when trying to access a wrong url path (https://virginiatest.dev.renku.ch/projects/abcdefg)
![Screenshot from 2019-08-27 13-30-12](https://user-images.githubusercontent.com/43481553/63767985-54eb4100-c8cf-11e9-9010-1a01d3ce7c60.png)

vfried commented:
> This has already been approved, but I found a couple of edge cases that may be confusing for the users
> 
> * Endless spinning wheel when trying to access a non existing project by id -- or a private project as well (https://virginiatest.dev.renku.ch/projects/9500/)
>   ![Screenshot from 2019-08-27 13-29-48](https://user-images.githubusercontent.com/43481553/63767974-50bf2380-c8cf-11e9-9ac9-3ca963ff83e4.png)
> * Empty project when trying to access a wrong url path (https://virginiatest.dev.renku.ch/projects/abcdefg)
>   ![Screenshot from 2019-08-27 13-30-12](https://user-images.githubusercontent.com/43481553/63767985-54eb4100-c8cf-11e9-9010-1a01d3ce7c60.png)

Ill fix this before we push it...

vfried commented:
I fixed the not found pages, there where some errors and also some new pages added. I think they could have been part of a new issue but i wasn't sure, since is a small change i added it here. 

If you navigate to a path that doesn't exist inside a project you see this:
![image](https://user-images.githubusercontent.com/42647877/63776336-7ce3a000-c8e1-11e9-9b8a-747ea4c2c1a3.png)

If you navigate to a path under /projects that doesn't exist you see this: 
![image](https://user-images.githubusercontent.com/42647877/63776385-97b61480-c8e1-11e9-9f96-ac5981e4366d.png)

If you navigate to a project with a numeric id that doesn't exist:
![image](https://user-images.githubusercontent.com/42647877/63776478-c46a2c00-c8e1-11e9-9dbb-2d9256fd9291.png)

rokroskar commented:
This doesn't seem like a huge problem but when I go to a URL for a blob directly, I see the 404 page flash for a split second... Then the correct page with the file content loads.

vfried commented:
> This doesn't seem like a huge problem but when I go to a URL for a blob directly, I see the 404 page flash for a split second... Then the correct page with the file content loads.

I see what you mean... I'll try to fix it.

lorenzo-cavazzi commented:
> I am trying to navigate on different pages.
> 
> I get an error when I try to visualize a notebook file while in "lineage view", like in this one
> https://virginiatest.dev.renku.ch/projects/lorenzo.cavazzi.tech/zurich-bikes-tutorial/files/lineage/notebooks/zhbikes-notebook.ipynb

info: this will be addressed by #582

vfried commented:
> * We have 2 different "Project not found" pages. Wouldn't it be better to have a single one?

I added a new not found page but is for when the project was found but the url inside the project wasn't found... this allows the user to still navigate inside the project. If we put a simple project not found page the user can't navigate inside the project and it would also be technically wrong since the project is there.

rokroskar commented:
Now that this is in place, we should discuss the larger topic of providing URIs for Renku entities. This should be in-sync with how entities are identified in the KG.
